flink-kubernetes/src/main/java/org/apache/flink/kubernetes/utils/KubernetesUtils.java

```diff
@@ -487,6 +489,22 @@ public static <T> String resolveUserDefinedValue(
         return resolvedValue;
     }
 
+    /**
+     * Resolve the DNS policy defined value. First check if there is a DNS Policy override in
+     * PodTemplate, if not, check if host network is enabled.
+     *
+     * @param dnsPolicy DNS Policy defined in PodTemplate spec
+     * @param hostNetworkEnabled Host Network enabled
+     * @return the resolved value
+     */
+    public static String resolveDNSPolicy(String dnsPolicy, boolean hostNetworkEnabled) {
```

Comment:
Given that the Flink configuration options have a higher priority, I think `DNS_POLICY_HOSTNETWORK` need to be used if the host network enabled by `kubernetes.hostnetwork.enabled` explicitly.

And we need a test to guard this behavior.